### PR TITLE
Window Scent Block Review

### LINKED ISSUE #48300: Taped windows don't reduce scent dispersion as much as a regular closed window
# Describe the bug

Taped Windows don't reduce scent dispersion.

# Steps To Reproduce

Be next to a taped up window, your scent cloud will travel through it as if it was an open window.

# Expected behavior

Being the same object, the latter simply having duct tape applied to it, It should reduce your scent cloud at the same level as a normal closed window.

# Screenshots

Screenshot (59)

# Versions and configuration

 - OS: Windows 10
 - Game Version: 0.E-10255-g6b8dd00
 - Graphics version: Tiles

# Additional context

The taped up windows, along with stained glass windows, lack the "REDUCE_SCENT" flag in terrain-windows.json.

### PR DESCRIPTION
**Pull request by souricelle**
#### Summary
Bugfixes "Window Scent Block Review"

#### Purpose of change

It was discovered in #48300 that taped windows weren't blocking scent, which seemingly defeated the point of them. On further review, I discovered that the scent_block and blocks_wind flags were missing from many windows and applied to many windows that shouldn't have them, such as windows with no glass at all. In fact, taping a window up was _removing_ the window's ability to block scent. This PR amends every window in the game to follow the same simple guidelines for scent and wind blocking.
 
#### Describe the solution

A window blocks wind if it:

* Has a glass pane that is shut 
* Has a plastic pane that is shut
* Is reinforced so that it's basically a wall

A window blocks scent and wind if it:

* Is double (or more) glazed with glass or plastic, or uses higher quality construction IE tempered glass or stained glass.
* Has glass or plastic in it and is taped or boarded up 
* Is boarded up and reinforced so that it's basically a wall, even if there's no glass.

Single pane windows (such as most of the windows in houses you come across) block wind but not scent. If the window glass is broken whether boarded or not it blocks neither. If it is intact and boarded, it blocks both. If it is broken or intact but is boarded and reinforced, it blocks both.

Fixes #48300 

Barred windows (such as those on electronics stores) block scent and wind as long as the glass is intact. Once it shatters it they block neither.

#### Describe alternatives you've considered

We could create a unique boarded up window type for every single kind of window in the game so that single-pane windows don't become scent-blocking until boarded up and reinforced, but that sounds like unnecessary bloat. It seems better to assume that the work the survivor is doing to secure the intact window is enough to help keep scents inside. The windows are not airtight in any case so this doesn't seem too weird to me.

#### Testing

I enabled scent visualization and created a huge building with every type of window in the game in it. I stood in front of each one to check whether my scent was reduced by the window appropriately or not.

#### Additional context

There are a ton of grammar errors and what appear to be some actual glitches in windows, these will be tackled separately but this seems good enough for a simple patch during string freeze.

### CODE REVIEW

File: data/json/furniture_and_terrain/terrain-windows.json [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Cataclysm: Dark Days Ahead JSON Web Linting Tool
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/terrain-windows.json:1]: Remove this.

**Comment by souricelle** [comment]
Something weird is going on, I'm seeing

-    "curtain_transform": "t_reinforced_quadruple_pane_glass",
+    "curtain_transform": "t_reinforced_quadruple_pane_glass_open",

and some other changes that I didn't make. Did the linting tool mess with my changes?

**Comment by souricelle** [comment]
Getting an error, "opening terrain t_metal_grate_window_with_curtain_open for t_metal_grate_window_with_curtain_noglass doesn't reciprocate" in mapdata.cpp line 1295.

This error didn't happen on my machine until I used the Linting tool, but I don't see that the linting tool changed anything about those windows. Any ideas?

**Comment by actual-nh** [comment]
> Getting an error, "opening terrain t_metal_grate_window_with_curtain_open for t_metal_grate_window_with_curtain_noglass doesn't reciprocate" in mapdata.cpp line 1295.
> 
> This error didn't happen on my machine until I used the Linting tool, but I don't see that the linting tool changed anything about those windows. Any ideas?

Seems to be working now? "Fixes #48300" would be good to add to the PR, BTW.

**Comment by souricelle** [comment]
The bug was apparently something on my end due to the fiddling around I was doing mid-save.

This PR also fixes an additional bug I found with reinforced plastic windows, which apparently nobody was using or they'd have noticed too.

**Comment by souricelle** [comment]
I need to fix something in the JSONs, it looks like some of the old windows (that I didn't touch) had bad flags but the game wasn't complaining about them until I ran the linting tool.

**Comment by souricelle** [comment]
OK, I got it. There were some very old bugs on a couple of the less-common windows (reinforced plastic and barred reinforced) where they had improper open/close flags that didn't reciprocate. Somehow it wasn't throwing errors until the linter tidied up the formatting. The whole thing is now bug free and ready to be merged.

File: data/json/furniture_and_terrain/terrain-windows.json [hunk 2]
```diff
@@ -246,7 +235,7 @@
       "SMALL_PASSAGE",
       "WINDOW"
     ],
-    "curtain_transform": "t_window_no_curtains_open",
+    "curtain_transform": "t_window_no_curtains",
```
Comment (ama0) [inline data/json/furniture_and_terrain/terrain-windows.json:238]: This should be reverted.

**Comment by ama0** [comment]
`"curtain_transform:"` describes the new kind of window a window transforms into when removing the curtains.

* A **closed** window _with_ curtains should transform into a **closed** window _without_ curtains.
* An **open** window _with_ curtains should transform into an **open** window _without_ curtains.

Commit https://github.com/CleverRaven/Cataclysm-DDA/pull/48323/commits/c9cff198c72028ad8e6431bc48be5b9d03efb462  (Fix old bugs) above changed it so open windows transform into closed windows when removing curtains.

IMO most of https://github.com/CleverRaven/Cataclysm-DDA/pull/48323/commits/c9cff198c72028ad8e6431bc48be5b9d03efb462  should be reverted

**Pull request merged**